tests/components/remootio/test_cover.py

```diff
@@ -0,0 +1,433 @@
+"""Test the Remootio cover entity."""
+import logging
+from unittest.mock import AsyncMock, MagicMock, Mock, PropertyMock, patch
+
+from aioremootio import (
+    Event,
+    EventSource,
+    EventType,
+    Listener,
+    RemootioClient,
+    State,
+    StateChange,
+)
+
+from homeassistant.components.cover import DOMAIN as COVER_DOMAIN, CoverDeviceClass
+from homeassistant.components.remootio.const import (
+    CONF_API_AUTH_KEY,
+    CONF_API_SECRET_KEY,
+    CONF_SERIAL_NUMBER,
+    DOMAIN,
+)
+from homeassistant.const import (
+    CONF_DEVICE_CLASS,
+    CONF_HOST,
+    STATE_CLOSED,
+    STATE_CLOSING,
+    STATE_OPEN,
+    STATE_OPENING,
+)
+from homeassistant.core import HomeAssistant
+
+from tests.common import MockConfigEntry, async_capture_events
+
+TDV_HOST = "127.0.0.1"
+TDV_CREDENTIAL = "123456789A123456789B123456789C123456789D123456789E123456789FVXYZ"
+TDV_SERIAL_NUMBER = "1234567890"
+TDV_TITLE = "remootio"
+TDV_ENTITY_ID = "cover.remootio"
+
+_LOGGER = logging.getLogger(__name__)
+
+remootio_client_state_change_listeners: list[Listener[StateChange]] = []
+remootio_client_event_listeners: list[Listener[Event]] = []
+remootio_client: Mock = None
+
+
+@patch("homeassistant.components.remootio.create_client")
+async def test_open_when_closed(remootio_create_client: Mock, hass: HomeAssistant):
+    """Tests the opening of the Remootio controlled device."""
+    global remootio_client
```

Comment:
no global objects during tests. they linger forever. Add a fixture that return an client instance, that could be set to module scope if you want it to live during all tests, but try to keep each test separate.

Reply:
```python
@pytest.fixture(name="remootio_client", autouse=True)
def remootio_client_fixture():
  with patch("homeassistant.components.remootio.create_client") as client:
    client.return_value = _initialize_remootio_client()
    yield client.return_value
```

Then if you need the client object, you can just add remootio_client as a parameter to your test function.